core-bundle/src/Search/Indexer/DelegatingIndexer.php

```diff
@@ -30,22 +30,61 @@ public function addIndexer(IndexerInterface $indexer): self
 
     public function index(Document $document): void
     {
-        foreach ($this->indexers as $indexer) {
-            $indexer->index($document);
-        }
+        $this->wrapInTryCatch(
+            static function (IndexerInterface $indexer) use ($document): void {
+                $indexer->index($document);
+            },
+        );
     }
 
     public function delete(Document $document): void
     {
-        foreach ($this->indexers as $indexer) {
-            $indexer->delete($document);
-        }
+        $this->wrapInTryCatch(
+            static function (IndexerInterface $indexer) use ($document): void {
+                $indexer->delete($document);
+            },
+        );
     }
 
     public function clear(): void
     {
+        $this->wrapInTryCatch(
+            static function (IndexerInterface $indexer): void {
+                $indexer->clear();
+            },
+        );
+    }
+
+    private function wrapInTryCatch(callable $function): void
+    {
+        $warningsOnly = true;
+        $indexerExceptions = [];
+
         foreach ($this->indexers as $indexer) {
-            $indexer->clear();
+            try {
+                $function($indexer);
+            } catch (IndexerException $exception) {
+                $indexerExceptions[] = $exception;
+                if (!$exception->isOnlyWarning()) {
+                    $warningsOnly = false;
+                }
+            } catch (\Throwable) {
+                // noop
```

Comment:
Shouldn't we still throw here? Otherwise other errors will just get lost.

Reply:
We can also collect them but we must not throw otherwise the other indexers are never called.

Reply:
Addressed in https://github.com/contao/contao/pull/8342/commits/ad469a352fd569fceb8540111a6ba03d4a56eba2.

Reply:
My thinking was that if the exception is not an `IndexerException`, some other fatal error occurred and then it would be ok to not call the other indexers anymore.

Reply:
I disagree. If I am registering an indexer I want it to be called - I cannot influence the code of the other indexers - that's not my responsibility.

Reply:
I understand - but the issue with the current solution is, that it will still hide the original exception. Having only the exception message will not be useful without the original exception and its previous execptions and their stack traces.

Reply:
For example, in our `Cron` we simply throw the first exception for that reason.